taxi/views.py

```diff
@@ -1,47 +1,53 @@
+from django.contrib.auth.decorators import login_required
+from django.contrib.auth.mixins import LoginRequiredMixin
 from django.shortcuts import render
 from django.views import generic
 
 from .models import Driver, Car, Manufacturer
 
 
+@login_required
 def index(request):
     """View function for the home page of the site."""
 
     num_drivers = Driver.objects.count()
     num_cars = Car.objects.count()
     num_manufacturers = Manufacturer.objects.count()
+    num_visits = request.session.get("num_visits", 1)
```

Comment:
Ensure that the `num_visits` variable works as expected. According to the checklist, when you visit the page for the first time, it should display: `You have visited this page 1 time`.